lib/Sema/TypeCheckAttr.cpp

```diff
@@ -3885,8 +3885,20 @@ void AttributeChecker::visitDerivativeAttr(DerivativeAttr *attr) {
   auto vectorTy = valueResultConf.getTypeWitnessByName(
       valueResultType, Ctx.Id_TangentVector);
 
+  // Compute the actual differential/pullback type that we use for comparison
```

Comment:
Note: https://github.com/apple/swift/pull/28738 upstreams `@derivative` type-checking to `master`. I can handle cherry-picking this fix to `master` after that PR is merged.

I wonder if it's possible to write a unit test that can be upstreamed with the fix?

Food for thought: after code is upstreamed, perhaps we should make changes on `master` and merge them (or let them be merged) to `tensorflow`. This saves some effort compared with committing to both `tensorflow` and `master`, since code must eventually land in `master`.